custom_components/ai_thermostat/climate.py

```diff
@@ -70,823 +67,765 @@
+				
+				# window open detection and weather detection force turn TRV off
+				if (self.window_open or not is_cold) and not self.closed_window_triggered:
+					self.heating_active_pre_window_opened = False
+					if self._hvac_mode == HVAC_MODE_HEAT:
+						self.heating_active_pre_window_opened = True
+					self._hvac_mode = HVAC_MODE_OFF
+					self.closed_window_triggered = True
+				elif self.heating_active_pre_window_opened:
+					self._hvac_mode = HVAC_MODE_HEAT
+				
+				try:
+					remapped_states = convert_outbound_states(self, self._hvac_mode)
+					converted_hvac_mode = remapped_states.get('system_mode')
+					current_heating_setpoint = remapped_states.get('current_temperature')
+					has_real_mode = True if self.hass.states.get(self.heater_entity_id).attributes.get('system_mode') is not None else False
+					calibration = remapped_states.get('calibration')
+					
+					# if off do nothing
+					if self.closed_window_triggered or self._hvac_mode == HVAC_MODE_OFF:
+						if has_real_mode:
+							if self.hass.states.get(self.heater_entity_id).attributes.get('system_mode') == HVAC_MODE_OFF:
+								self.ignore_states = False
+								return
+						if self.calibration_type == 1:
+							if float(self.hass.states.get(self.heater_entity_id).attributes.get('current_heating_setpoint')) <= 5.0:
+								self.ignore_states = False
+								return
+					
+					# Only send the local_temperature_calibration if not instantly following
+					do_calibration = False
+					if (datetime.now() > (self.last_calibration + timedelta(seconds=20))):
+						do_calibration = True
+						self.last_calibration = datetime.now()
+					
+					_LOGGER.debug(
+							"ai_thermostat triggered states > window open: %s night mode: %s Mode: %s set: %s has_mode: %s Calibration: %s - send: %s set_temp: %s cur_temp: %s Model: %s Calibration "
+							"type: %s winter: %s TRV: %s",
+							self.window_open,
+							self.night_status,
+							converted_hvac_mode,
+							self._hvac_mode,
+							has_real_mode,
+							calibration,
+							do_calibration,
+							current_heating_setpoint,
+							self._cur_temp,
+							self.model,
+							self.calibration_type,
+							is_cold,
+							self.hass.states.get(self.heater_entity_id).attributes.get('device').get('friendlyName')
+					)
+					
+					if self.calibration_type == 1 and float(self.hass.states.get(self.heater_entity_id).attributes.get('current_heating_setpoint')) != float(calibration):
+						current_heating_setpoint = calibration
+						mqtt_settemp = {"current_heating_setpoint": float(calibration)}
+						payload = json.dumps(mqtt_settemp, cls=JSONEncoder)
+						await self.mqtt.async_publish(self.hass, 'zigbee2mqtt/' + self.hass.states.get(self.heater_entity_id).attributes.get('device').get('friendlyName') + '/set', payload, 0, False)
+						data = {ATTR_ENTITY_ID: self.heater_entity_id, "temperature": float(calibration)}
+						await self.hass.services.async_call('climate', SERVICE_SET_TEMPERATURE, data)
```

Comment:
@RubenKelevra This is an example for the HA direct service call, the SERVICE_SET_TEMPERATURE and SERVICE_SET_HVAC_MODE only accept set temp and mode, since HA 2021.12 the local_calibration endpoint is shared in HA as number input, thats nice it must be possible to set it with the number service, but i need to figure out how to get the heater entity related entities to acces this

Reply:
Well, the naming scheme is fixed for the device types. So we could just configure an e.g. "*_position" variable for the TS0601 in the yaml

Reply:
Yes its outdated Code, i did Changes to do such a thing. but the Problem is, its renameable in ha

Reply:
Who would rename the position variable of his thermostats? That's more than just an edge case 😆

Reply:
i didn't mean the attr, i rewrite it to use the HA services, for valve pos and local calib there are entities in HA and they can be renamed by the user. but all of them are linked to the climate devices.